doc/ug/getting_started_sw_testing.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Test Software
+---
+
+## Prerequisites
+
+_Make sure you followed the install instructions to [prepare the system]({{< relref "install_instructions#system-preparation" >}}) and to install the [software development tools]({{< relref "doc/ug/install_instructions#software-development" >}}) and [Verilator]({{< relref "install_instructions#verilator" >}})._
+
+## Overview
+
+The OpenTitan software tests fall into two main categories:
+
+* Unit tests
+* Smoke tests
+
+Unit tests are designed to run on the host so they are quick to build and run.
+Writing unit tests is a good way to provide high levels of test coverage for a library without significantly impacting the runtime of the continuous integration tests.
+Smoke tests on the other hand are designed to run on an emulated device and therefore require a more complicated setup procedure.
+Typically smoke tests take significantly longer to run than unit tests.
+Smoke tests are therefore best suited for testing that the core functionality of a library is correct.
+
+## Unit tests
+
+OpenTitan unit tests are written using the [GoogleTest](https://github.com/google/googletest/) C++ test framework and are designed to be run on a workstation.
+Typically this means that they are executed in a Linux environment running on the x86_64 architecture although they should be designed to be as portable as possible.
+Device-specific features that are not necessarily available on the host system, for example memory-mapped I/O (MMIO) and Ibex control registers (CSRs), may be provided by mock implementations.
+
+Unit tests use the suffix `_unittest.cc` and are typically contained in the same directory as other tests for the same code.
+
+The unit tests may be executed by invoking the ninja test target:
+
+```console
+# Configure the Meson environment.
+$ cd "$REPO_TOP"
+$ ./meson_init.sh
+
+# Build and run the unit tests.
+$ ninja -C build-out test
+```
+
+## Device smoke tests
+
+OpenTitan tests that are designed to be executed on the target device itself are referred to as ‘smoke tests’.
+Smoke tests are full applications written in a language suitable for the target device such as C.
+The target device may be emulated using either software simulation or a programmable hardware device (FPGA).
```

Comment:
"Emulation" is a word everybody defines differently (and then fights about being right). I'd just skip the word.